Packs/QuestKace/Integrations/QuestKace/QuestKace.py

```diff
@@ -0,0 +1,907 @@
+    Returns:
+        list of queue ids.
+    """
+    if queue:
+        queues_id = queue
+    else:
+        queues_id = get_queue_ids(client)
+    fields: list = []
+    for q in queues_id:
+        client.update_token()
+        fields_by_queue = client.queues_list_fields_request(queue_number=str(q))
+        fields_by_queue = fields_by_queue.get('Fields', [])
+        for field in fields_by_queue:
+            if field.get('jsonKey') not in fields:
+                # get internal error 500 from server with related tickets
+                if field.get('jsonKey') != 'related_tickets' and field.get('jsonKey') != 'referring_tickets':
+                    fields.append(field.get('jsonKey'))
+    return fields
+
+
+def get_queue_ids(client: Client) -> list:
+    """
+    Creating a list of all queue ids that are in the system.
+    Args:
+        client: Client for the api.
+    Returns:
+        list of queue ids.
+    """
+    queues = client.queues_list_request()
+    queues = queues.get('Queues', [])
+    queues_id = []
+    for q in queues:
+        queues_id.append(str(q.get('id')))
+    return queues_id
+
+
+def shaping_by_fields(fields: list) -> str:
+    """
+    Creating a shaping for the request which is from the fields and seperated by comma's
+    Args:
+        fields: List of fields that would be part of the shaping.
+    Returns:
+        str of the shaping.
+    """
+    shaping = 'hd_ticket all'
+    for field in fields:
+        shaping += f',{field} limited'
+    return shaping
+
+
+def set_shaping(client, queue: Optional[list] = None) -> str:
+    """
+    Creating a shaping for the request.
+    Args:
+        client: Client in order to get the queue fields.
+        queue: If specific queue is given for the shaping.
+    Returns:
+        str of the shaping.
+    """
+    fields = get_fields_by_queue(client, queue)
+    shaping = shaping_by_fields(fields)
+    return shaping
+
+
+def parse_incidents(items: list, fetch_limit: str, time_format: str, parsed_last_time: datetime) \
+        -> Tuple[list, Any]:
+    """
+    This function will create a list of incidents
+    Args:
+        items : List of tickets of the api response.
+        fetch_limit: Limit for incidents of fetch cycle.
+        time_format: Time format of the integration.
+        parsed_last_time: limit for number of fetch incidents per fetch.
+    Returns:
+        incidents: List of incidents.
+        parsed_last_time: Time of last incident.
+    """
+    count = 0
+    incidents = []
+    for item in items:
+        if count >= int(fetch_limit):
+            break
+
+        incident_created_time = dateparser.parse(item['created'])
+
+        incident = {
+            'name': item['title'],
+            'occurred': incident_created_time.strftime(time_format),
+            'rawJSON': json.dumps(item)
+        }
+
+        incidents.append(incident)
+        count += 1
+        parsed_last_time = incident_created_time
```

Comment:
do we have to set it every time? why not just take it from the last item in items outsude of the loop?
also, let's rename it something else throughout the code, to imply that it came from an incident. e.g: `last_incident_time`

Reply:
we do not have to set it every time , The last one of the loop is already converted to a string from date object and I expect during the whole functions date object. So this requires the time of all the fetch function.